runners/direct-java/src/main/java/org/apache/beam/runners/direct/portable/ReferenceRunner.java

```diff
@@ -110,11 +112,28 @@ static ReferenceRunner forInProcessPipeline(RunnerApi.Pipeline p, Struct options
     return GreedyPipelineFuser.fuse(withGbks).toPipeline();
   }
 
+  private static Set<PCollectionNode> getKeyedPCollections(
+      ExecutableGraph<PTransformNode, PCollectionNode> graph) {
+    // This mimics KeyedPValueTrackingVisitor behavior in regular direct runner,
+    // but without propagating keyed-ness through key-preserving DoFn's.
+    // That is not yet necessary, but will be necessary once we implement state and timers.
```

Comment:
Can you add a bug that calls this out and reference it here?

Reply:
Done.